src/prefect/core/flow.py

```diff
@@ -149,12 +149,11 @@ def __init__(
     ):
         self._cache = {}  # type: dict
 
-        self.logger = logging.get_logger("Flow")
-
         if not name:
             raise ValueError("A name must be provided for the flow.")
 
         self.name = name
+        self.logger = logging.get_logger(self.name)
```

Comment:
This might make it difficult to distinguish between task and flow logs with ambiguous names -- perhaps the flow's logger should be like this:
```suggestion
        self.logger = logging.get_logger("Flow: {}".format(self.name))
```

Reply:
Ah good point; for easier querying maybe I should do `'Flow.{}'.format(self.name)`

Reply:
Since the name itself will arbitrary text (even... dareisay... emojis), I don't think we need to formalize the pattern too much

Reply:
Good call - I updated per your suggestion in my own commit (since I also needed to update tests).